From d26c9831a59b756d0468f918996399570f4737cb Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Thu, 14 Jul 2016 06:26:13 -0400
Subject: [PATCH] Prevent possible buffer overflow when reading TIFF images
 (bug report from Shi Pu of MS509 Team)

(cherry picked from commit c20de102cc57f3739a8870f79e728e3b0bea18c0)

This fix CVE-2016-5010

Origin; upstream, http://git.imagemagick.org/repos/ImageMagick/commit/c20de102cc57f3739a8870f79e728e3b0bea18c0
bug-debian: https://bugs.debian.org/832968
---
 coders/tiff.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/coders/tiff.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/tiff.c	2016-11-15 15:44:07.173787048 -0500
+++ imagemagick-6.6.9.7/coders/tiff.c	2016-11-15 15:44:07.173787048 -0500
@@ -1222,7 +1222,8 @@
               quantum_type=GrayQuantum;
               pad=(size_t) MagickMax((size_t) samples_per_pixel-1,0);
             }
-        status=SetQuantumPad(image,quantum_info,pad*((bits_per_sample+7) >> 3));
+        status=SetQuantumPad(image,quantum_info,pad*pow(2,ceil(log(
+          bits_per_sample)/log(2))));
         if (status == MagickFalse)
           {
             TIFFClose(tiff);
